### ISSUE TITLE
.

### ISSUE THREAD

**Issue by 87sam12** [keras-team/tf-keras#244]
Please go to TF Forum for help and support:

https://discuss.tensorflow.org/tag/keras

If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.11.0
- Python version: 3.8.10
- Bazel version (if compiling from source): 
- GPU model and memory: Not using any
- Exact command to reproduce: 
```py
model_checkpoint_callback = keras.callbacks.ModelCheckpoint(
    filepath=checkpoint_filepath,
    save_weights_only=False,
    monitor='accuracy',
    mode='max',
    save_best_only=True,
    save_freq=100,
    verbose=1)

model.fit(x=X, y=y, batch_size=100, epochs=200, verbose=2, callbacks=[model_checkpoint_callback])
```

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"

**Describe the problem**.

RAM memory is getting exhausted after several epochs before completing entire training

**Describe the current behavior**.
Basically, lets say I have 100 epochs, my chkpoints get created every 25. Then by 51st epoch my ram gets full. Now I want to continue training with same data to improve accuracy until the highest is achieved. I saw that my chkpoints model file size is very less (its around 10mb, and my ram is 4gb; then I just need my ipynb jupyter to run further after flushing out everything else if that 10mb file would have a cumulative representation of all past epochs training). Then why do I need the rest of the memory for? 

**Describe the expected behavior**.
I basically want to remove all the prior epochs temp data if any on ram. how to achieve this? I can basically remove any variables or other data and the run garbage collection. but how to clear variables if everything happens without any tangible variables? as model.fit method runs for all epochs and saves the periodical updates onto ram I suppose somewhere (unsure)

I found (https://keras.io/api/callbacks/model_checkpoint/) useful
I found other ways like:
```py
from memory_profiler import profile
```
but Im looking for something to alert me on a threshold.. these above tools just periodically print usage
for ram's mem usage.
there r other methods like
```py
import psutil

process = psutil.Process(os.getpid())
print(process.memory_info())
```
looking for threshold checks in place without failure of training

any ideas?

My training data does not use images.
How to exactly clear ram memory? more than that the local/global/that program related stuff must never be removed.. also will the further epochs be possible even after clearing ram? will the best model chkpoint saved be sufficient for further training/epochs?

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no):
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.

Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Source code / logs**.

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.
